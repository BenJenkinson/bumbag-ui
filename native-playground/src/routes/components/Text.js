import React from 'react';
import { Box, Text } from 'bumbag-native';
import { Preview } from '../../components/Preview';
import { PreviewSection } from '../../components/PreviewSection';

export default function App() {
  return (
    <Box>
      <Text fontSize="20px" fontWeight="500" marginBottom="16px">
        Text
      </Text>
      <PreviewSection title="Basic">
        <Preview>
          <Box>
            <Text use="strong">
              Hello world, this is a bit of <Text fontWeight="500">text</Text>.
            </Text>
          </Box>
        </Preview>
      </PreviewSection>
    </Box>
  );
}
