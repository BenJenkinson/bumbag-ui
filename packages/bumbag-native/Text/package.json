{
  "name": "bumbag-native/Text",
  "private": true,
  "sideEffects": false,
  "main": "../lib/Text",
  "module": "../es/Text",
  "types": "../ts/Text"
}