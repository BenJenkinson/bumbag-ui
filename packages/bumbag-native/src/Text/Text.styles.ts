import { Platform, Text } from 'react-native';
import { styled } from '../styled';
import { palette, theme } from '../utils/theme';

export const StyledText = styled(Text)`
  color: ${palette('text')};

  ${Platform.OS === 'web'
    ? `
      font-family: inherit;
      font-size: unset;
    `
    : ''}

  ${theme('native.Text', 'styles.base')};
`;
