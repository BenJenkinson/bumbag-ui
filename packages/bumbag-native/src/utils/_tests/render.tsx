import * as React from 'react';
import { ThemeConfig } from '../../types/theme';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (
  Component: any,
  { colorMode = 'default', theme = {} }: { colorMode?: string; theme?: ThemeConfig } = {}
) =>
  render(
    <Provider platform="native" theme={theme} colorMode={colorMode}>
      {Component}
    </Provider>
  );
