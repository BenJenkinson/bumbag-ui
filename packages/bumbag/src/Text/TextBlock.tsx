import { Box as ReakitBox } from 'reakit';

import { useClassName, createComponent, createElement, createHook } from '../utils';
import { Box, BoxProps } from '../Box';

import * as styles from './styles';

export type LocalTextBlockProps = {};
export type TextBlockProps = BoxProps & LocalTextBlockProps;

const useProps = createHook<TextBlockProps>(
  (props, { themeKey, themeKeyOverride }) => {
    const boxProps = Box.useProps(props);

    const className = useClassName({
      style: styles.TextBlock,
      styleProps: props,
      themeKey,
      themeKeyOverride,
      prevClassName: boxProps.className,
    });

    return { ...boxProps, className };
  },
  { themeKey: 'TextBlock' }
);

export const TextBlock = createComponent<TextBlockProps>(
  (props) => {
    const textBlockProps = useProps(props);
    return createElement({ children: props.children, component: ReakitBox, use: props.use, htmlProps: textBlockProps });
  },
  {
    attach: {
      useProps,
      displayName: 'TextBlock',
    },
    defaultProps: {
      use: 'span',
    },
    themeKey: 'TextBlock',
  }
);
