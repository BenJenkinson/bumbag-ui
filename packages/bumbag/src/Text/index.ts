import * as styles from './styles';

export * from './Text';
export { styles as textStyles };
