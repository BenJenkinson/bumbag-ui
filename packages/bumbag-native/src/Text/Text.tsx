import { TextProps as RNTextProps } from 'react-native';
import { createComponent, createElement, createHook } from 'bumbag/utils';

import { Box, BoxProps } from '../Box';
import * as styles from './Text.styles';

export type LocalTextProps = {};
export type TextProps = BoxProps & LocalTextProps;

const useProps = createHook<TextProps>(
  (props) => {
    const boxProps = Box.useProps(props);
    return { ...boxProps };
  },
  { themeKey: 'Text' }
);

export const Text = createComponent<RNTextProps & TextProps>(
  (props) => {
    const textProps = useProps(props);
    return createElement({
      children: props.children,
      component: styles.StyledText,
      htmlProps: textProps,
    });
  },
  {
    attach: {
      useProps,
      displayName: 'Text',
    },
    themeKey: 'Text',
  }
);
