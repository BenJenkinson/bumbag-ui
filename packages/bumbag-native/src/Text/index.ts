import * as textStyles from './Text.styles';

export * from './Text';
export { textStyles };
