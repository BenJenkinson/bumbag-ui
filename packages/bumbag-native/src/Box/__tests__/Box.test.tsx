import * as React from 'react';
import { Box } from '../index';
import { Text } from '../../Text';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Box />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should assign a ref', () => {
    const ref = React.createRef();
    render(
      <Box ref={ref}>
        <Text>Hello world</Text>
      </Box>
    );
    expect(ref.current).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Box backgroundColor="red" color="white" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it.todo('should render correctly with CSS props with breakpoints');

  it.todo('should render correctly with CSS props with pseudo props');

  it('should render correctly with a color CSS prop', () => {
    const { container } = render(<Box backgroundColor="white900" color="primary" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with a spacing CSS prop', () => {
    const { container } = render(<Box margin="major-4" paddingLeft="major-2" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
