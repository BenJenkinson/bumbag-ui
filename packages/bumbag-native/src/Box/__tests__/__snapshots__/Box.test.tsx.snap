// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should assign a ref 1`] = `
<div
  class="css-view-1dbjc4n"
>
  <div
    class="css-text-901oao"
    dir="auto"
    style="font-family: unset;"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
<div
  class="css-view-1dbjc4n"
/>
`;

exports[`props should render correctly with CSS props 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(255, 0, 0); color: rgb(255, 255, 255);"
/>
`;

exports[`props should render correctly with a color CSS prop 1`] = `
<div
  class="css-view-1dbjc4n"
  style="background-color: rgb(217, 217, 225); color: rgb(87, 79, 235);"
/>
`;

exports[`props should render correctly with a spacing CSS prop 1`] = `
<div
  class="css-view-1dbjc4n"
  style="margin: 2rem 2rem 2rem 2rem; padding-left: 1rem;"
/>
`;
